src/vl-wizard.js

```diff
@@ -0,0 +1,120 @@
+import '/lib/wizard.js';
+import './vl-wizard-pane.js';
+
+/**
+ * VlWizard
+ * @class
+ * @classdesc Gebruik de wizard om de gebruiker door een proces met meerdere stappen te begeleiden. Een wizard laat je toe om een complex proces op te splitsen in hapklare acties. Een wizard maakt het ook mogelijk om opties afhankelijk te maken van de keuzes in een voorgaande stap.
+ * 
+ * @extends VlElement
+ * 
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-wizard/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-wizard/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-wizard.html|Demo}
+ * 
+ */
+export class VlWizard extends VlElement(HTMLElement) {
+    constructor() {
+        super(`
+            <style>
+                @import '/src/style.css';
+
+                .vl-wizard__panes {
+                    overflow: hidden;
+                }
+            </style>
+            <section class="vl-wizard" data-vl-wizard>
+                <header class="vl-wizard__heading" role="none">
+                    <slot name="title"></slot>
+                    <slot name="header"></slot>
+                </header>
+                <vl-progress-bar></vl-progress-bar>
+                <div class="vl-wizard__panes">
+                    <slot name="panes"></slot>
+                </div>
+            </section>
+        `);
+
+        this.__callback = {
+            callbackFn: Promise.resolve(true)
+        };
+    }
+
+    async connectedCallback() {
+        await this._processPanes();
+        this._observeProgressBarClick();
+        this._dress();
+    }
+
+    /**
+     * Geeft de callback functie die bepaalt of er naar de volgende/vorige pagina mag genavigeerd worden.
+     * 
+     * @return {Function}
+     */
+    get callback() {
+        return this.__callback;
+    }
+
+    /**
+     * Callback setter die bepaalt of er naar de volgende/vorige pagina mag genavigeerd worden.
+     * 
+     * @param {Function} callback
+     */
+    set callback(callback) {
+        this.__callback.callbackFn = callback;
+    }
+
+    get _panes() {
+        return this.querySelectorAll('vl-wizard-pane');
+    }
+
+    get _activePane() {
+        return [... this._panes].find(pane => pane.isActive);
+    }
+
+    get _activePaneNumber() {
+        return [... this._panes].findIndex(pane => pane.isActive) + 1;
+    }
+
+    get _progressBar() {
+        return this._shadow.querySelector('vl-progress-bar');
+    }
+
+    _getProgressBarStepTemplate(content) {
+        return this._template(`
+            <vl-progress-bar-step>
+                ${content}
+            </vl-progress-bar-step>
+        `);
+    }
+
+    async _processPanes() {
+        this._panes.forEach(pane => {
+            this._progressBar.appendChild(this._getProgressBarStepTemplate(pane.title));
+        });
+    }
+
+    _dress() {
+        setTimeout(() => {
```

Comment:
Waarom is de timeout nodig?

Reply:
Om er zeker van te zijn dat al de elementen toegevoegd zijn die de dress functie van Webuniversum verwacht.

Reply:
Gaan we hierdoor geen race conditie introduceren bij het gebruik?
Een afnemer heeft momenteel geen manier om te bepalen of het dressed is of niet.
Kunnen we dan niet beter zorgen dat we met promises wachten om de dress te triggeren en die promise exposen naar afnemers oid.?

Reply:
Dat lijkt me niet de bedoeling, de dress gaat in dit geval de DOM niet wijzigen dus zie ik geen gevaar. De introductie van een ready getter analoog aan bijvoorbeeld vl-ui-select lijkt me niet nodig.